src/DialogTitle/DialogTitle.tsx

```diff
@@ -0,0 +1,47 @@
+import { Box, IconButton, DialogTitle as MuiDialogTitle } from "@mui/material";
+
+import CloseIcon from "@mui/icons-material/Close";
+import type { DialogTitleProps } from "./DialogTitle.types";
+import React from "react";
+
+/**
+ * This component is based on the material DialogTitle. It supports all the props that the material DialogTitle supports and adds extra functionality to show a close button.
+ */
+const DialogTitle = ({ children, onClose, ...other }: DialogTitleProps) => {
+  return (
+    <MuiDialogTitle
+      sx={{
+        display: "flex",
```

Comment:
Seems like this causes problems when there are no whitespaces in a string. Test it with the following string:
"displayyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy"

Then you'll see the following:
https://github.com/user-attachments/assets/e81e4a52-44f2-40bf-b3ab-3b0041e974e4

As you can see in the screen recording, this introduces a scrollbar in such a scenario